advanced/http-echo-operator/controllers/httpecho_controller.go

```diff
@@ -0,0 +1,222 @@
+			return ctrl.Result{}, err
+		}
+		// Deployment created successfully - return and requeue
+		return ctrl.Result{Requeue: true}, nil
+	} else if err != nil {
+		log.Error(err, "Failed to get Deployment")
+		return ctrl.Result{}, err
+	}
+
+	// Ensure the deployment size is the same as the spec
+	size := httpEcho.Spec.Size
+	if *found.Spec.Replicas != size {
+		found.Spec.Replicas = &size
+		err = r.Update(ctx, found)
+		if err != nil {
+			log.Error(err, "Failed to update Deployment", "Deployment.Namespace", found.Namespace, "Deployment.Name", found.Name)
+			return ctrl.Result{}, err
+		}
+		// Ask to requeue after 1 minute in order to give enough time for the
+		// pods be created on the cluster side and the operand be able
+		// to do the next update step accurately.
+		return ctrl.Result{RequeueAfter: time.Minute}, nil
+	}
+
+	// Update the HttpEcho status with the pod names
+	// List the pods for this HttpEcho's deployment
+	podList := &corev1.PodList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(httpEcho.Namespace),
+		client.MatchingLabels(labelsForHttpEcho(httpEcho.Name)),
+	}
+	if err = r.List(ctx, podList, listOpts...); err != nil {
+		log.Error(err, "Failed to list pods", "HttpEcho.Namespace", httpEcho.Namespace, "HttpEcho.Name", httpEcho.Name)
+		return ctrl.Result{}, err
+	}
+	podNames := getPodNames(podList.Items)
+
+	// Update status.Nodes if needed
+	if !reflect.DeepEqual(podNames, httpEcho.Status.Nodes) {
+		httpEcho.Status.Nodes = podNames
+		err := r.Status().Update(ctx, httpEcho)
+		if err != nil {
+			log.Error(err, "Failed to update HttpEcho status")
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *HttpEchoReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&httpv1alpha1.HttpEcho{}).
+		Complete(r)
+}
+
+// getPodNames returns the pod names of the array of pods passed in
+func getPodNames(pods []corev1.Pod) []string {
+	var podNames []string
+	for _, pod := range pods {
+		podNames = append(podNames, pod.Name)
+	}
+	return podNames
+}
+
+// labelsForHttpEcho returns the labels for selecting the resources
+// belonging to the given HttpEcho CR name.
+func labelsForHttpEcho(name string) map[string]string {
+	return map[string]string{"app": "http-echo", "http-echo_cr": name}
+}
+
+// serviceForHttpEcho returns a HttpEcho Deployment object
+func (r *HttpEchoReconciler) serviceForHttpEcho(m *httpv1alpha1.HttpEcho) *corev1.Service {
+	ls := labelsForHttpEcho(m.Name)
+
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      m.Name,
+			Namespace: m.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type:     corev1.ServiceTypeNodePort,
+			Selector: ls,
+			Ports: []corev1.ServicePort{
+				{Name: "http", Port: 8080, Protocol: corev1.ProtocolTCP, TargetPort: intstr.FromString("http-echo")},
```

Comment:
```suggestion
				{Name: "http", Port: 8080, NodePort: 30080, Protocol: corev1.ProtocolTCP, TargetPort: intstr.FromString("http-echo")},
```
This fixes the external port, so you don't have to look that up during destroy and apply of the instance